### ISSUE TITLE
New member state is pending, should be active when Admin adds user via profile widget

### ISSUE THREAD

**Issue by HopeNow** [Gizra/og#120]
**The Bug:**
When site administrator adds a user to a group (ours are invite only) using the widget on the user's profile page, the user's state is set to _pending_ instead of _active_.

**Context:**
1. Version: og 7.x-2.9
2. Likely caused by security update of og 2.9 in file "OgBehaviorHandler.class.php" on line 127 located in og/plugins/entityreference/behavior/ possibly because membership state provided by the widget is empty.
3. What's the best fix for this: Make the widget provide user's state? Or check if actor has administer group permission and set state to _active_ if they do?

**Culprit Code:**

```
    // Prepare an array with the membership state, if it was provided in the widget.
        ...
      // Must provide correct state in the event that approval is required.
      if (empty($item['state']) && $entity_type == 'user' && !og_user_access($group_type, $gid, 'subscribe without approval', $entity)) {
        $item['state'] = OG_STATE_PENDING;
      }...

```

**Slow Work around:**
Site administrator must add user to groups one by one from each group's add user page so that user's states are correctly set to _active_.

**Comment by alex-bukach** [comment]
> Or check if actor has administer group permission and set state to active if they do?

This options seems more reasonable to me, implemented in https://github.com/Gizra/og/pull/124.

**Comment by zerolab** [comment]
This was fixed in #124 and can be closed
